kmod/core/Makefile

```diff
@@ -1,24 +1,29 @@
 # make rules
 KPATCH_BUILD ?= /lib/modules/$(shell uname -r)/build
 KERNELRELEASE := $(lastword $(subst /, , $(dir $(patsubst %/,%,$(KPATCH_BUILD)))))
+VERCODE := $(shell awk '/LINUX_VERSION_CODE/{print $$NF}' $(KPATCH_BUILD)/include/generated/uapi/linux/version.h)
 THISDIR := $(abspath $(dir $(lastword $(MAKEFILE_LIST))))
 
 ifeq ($(wildcard $(KPATCH_BUILD)),)
 $(error $(KPATCH_BUILD) doesn\'t exist.  Try installing the kernel-devel-$(KERNELRELEASE) RPM or linux-headers-$(KERNELRELEASE) DEB.)
 endif
 
+ifeq ($(shell if [ $(VERCODE) -lt 328192 ];then echo "y";fi),y)
```

Comment:
Not feeling strongly about this, but you could define a:
`VERCODE_5_2 := 328192`
To clarify what that value is. (Arguably you can guess from the 'else' message, but I feel it reduces ambiguity).

Reply:
Geez, `make` makes it difficult to do something simple like compare two integers.  I'd almost be tempted to tack this comparison into the awk command, let it print "kpatch.ko" and then save the output into a $TARGETS variable... but I'm not sure that's any more or less elegant than the ifeq shell trick.  Either way, setting up a `VERCODE_5_2` variable might save somebody time if they aren't paying close attention to the comment that Julien mentioned.

Reply:
I think it's documented by the info line. I had commented it, but removed it would clutter more than anything.

Reply:
> Geez, `make` makes it difficult to do something simple like compare two integers. I'd almost be tempted to tack this comparison into the awk command, let it print "kpatch.ko" and then save the output into a $TARGETS variable... but I'm not sure that's any more or less elegant than the ifeq shell trick. Either way, setting up a `VERCODE_5_2` variable might save somebody time if they aren't paying close attention to the comment that Julien mentioned.

Oh, and for that. I could not agree more. The right way to do it in my opinion is NOT by using GNU make extensions but my having an external shell script that does the work and simply call that from the Makefile. At the moment it seems out of place. But if we use 'makeit', then this is certainly how that kind of logic is isolated from target definition.